internal/idp/providers/saml/saml.go

```diff
@@ -104,6 +108,36 @@ func WithEntityID(entityID string) ProviderOpts {
 	}
 }
 
+// ParseMetadata parses the metadata with the provided XML encoding and returns the EntityDescriptor
+func ParseMetadata(metadata []byte) (*saml.EntityDescriptor, error) {
```

Comment:
Could you add a test for the EntitiesDescriptor part? So that we have it covered as well in future changes.

Reply:
done